### PR TITLE
Log and Sleep task implementation

### PR DESCRIPTION
**Pull request by PhilipVinc**

### CODE REVIEW

**Comment by MatthieuDartiailh** [comment]
Just as a side note you do not need to create a new PR you can simply force push to the branch of the PR.

**Comment by MatthieuDartiailh** [comment]
As noticed by Codecov tests are not covering the case of a failing check because the time is < 0

**Comment by PhilipVinc** [comment]
The test for "test_sleep_task" actually passes. I have no idea what is causing the build not to pass.

**Comment by MatthieuDartiailh** [comment]
The failure is unrelated to your changes, I should try to make that test more robust to timeouts. When Travis is busy it becomes really slow sometimes.

File: ecpy/tasks/tasks/util/log_task.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# -----------------------------------------------------------------------------
+# Copyright 2015 by Ecpy Authors, see AUTHORS for more details.
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""Logging Task.
+
+"""
```
Comment (MatthieuDartiailh) [inline ecpy/tasks/tasks/util/log_task.py]: missing __future__ imports

File: ecpy/tasks/tasks/util/views/log_view.enaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright 2015 by Ecpy Authors, see AUTHORS for more details.
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""View for the Sleep Task.
```
Comment (MatthieuDartiailh) [inline ecpy/tasks/tasks/util/views/log_view.enaml]: wrong docstring

File: ecpy/tasks/tasks/util/sleep_task.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""Task that makes the system wait on all multithreading pools
+for a set amount of time.
+
+"""
+
```
Comment (MatthieuDartiailh) [inline ecpy/tasks/tasks/util/sleep_task.py]: \_\_future\_\_

File: tests/tasks/tasks/util/__init__.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
+
```
Comment (MatthieuDartiailh) [inline tests/tasks/tasks/util/__init__.py]: missing header and docstring

File: tests/tasks/tasks/util/test_log_task.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import pytest
+import enaml
+from multiprocessing import Event
+
+from ecpy.tasks.base_tasks import RootTask
+from ecpy.tasks.tasks.util.log_task import LogTask
+with enaml.imports():
+    from ecpy.tasks.tasks.util.views.log_view import LogView
+
+from ecpy.testing.tasks.util import CheckTask
```
Comment (MatthieuDartiailh) [inline tests/tasks/tasks/util/test_log_task.py]: unused

File: tests/tasks/tasks/util/test_log_task.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+class TestSleepTask(object):
+    """Test LogTask.
+
+    """
+
+    def setup(self):
+        self.root = RootTask(should_stop=Event(), should_pause=Event())
+        self.task = LogTask(name='Test')
+        self.root.add_child_task(0, self.task)
+        self.check = CheckTask(name='check')
```
Comment (MatthieuDartiailh) [inline tests/tasks/tasks/util/test_log_task.py]: useless

File: tests/tasks/tasks/util/test_log_task.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+    def test_perform1(self):
+        """Test checking that the message value gets written to the database
+
+        """
+        self.task.message = 'TestMessage'
```
Comment (MatthieuDartiailh) [inline tests/tasks/tasks/util/test_log_task.py]: would be nice to test for formatting

File: tests/tasks/tasks/util/test_sleep_task.py [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        assert not traceback
+        assert self.task.get_from_database('Test_time') == 2
+
+    def test_check2(self):
+        """Test handling a wrong string in the 'time' field.
+
+        """
+        self.task.time = 'a1.0'
+
+        test, traceback = self.task.check(test_instr=True)
```
Comment (MatthieuDartiailh) [inline tests/tasks/tasks/util/test_sleep_task.py]: keyword is useless

File: tests/tasks/tasks/util/test_log_task.py [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    """Test LogTask.
+
+    """
+
+    def setup(self):
+        self.root = RootTask(should_stop=Event(), should_pause=Event())
+        self.task = LogTask(name='Test')
+        self.root.add_child_task(0, self.task)
+        self.check = CheckTask(name='check')
+
```
Comment (MatthieuDartiailh) [inline tests/tasks/tasks/util/test_log_task.py]: We could also have a check for non-formattable string

**Pull request merged**